### DISCUSSION TITLE
Saving results as CSV from linker

### DISCUSSION THREAD

**Discussion by jmacak-at-dl** [moj-analytical-services/splink#1421] [Q&A] [answered]
I might be totally missing some part of the documentation that covers this, but is there a way to save the prediction and clustering tables as CSVs? You could do `clustering.as_pandas_dataframe().to_csv('path.csv')`, but that seems like an unnecessarily high-memory option when the tables are already in memory in DuckDB. I'd rather just persist the DuckDB table, which at this point is named something like `__splink__df_representatives_4125c69a1`, through some method like `SplinkDataFrame.to_csv('path.csv')`. Am I thinking about this the wrong way?

**Comment by ADBond** [comment, marked as answer]
There is a method `SplinkDataFrame.to_csv(filepath, overwrite=False)` which may be what you are after? Apologies - this appears to be currently undocumented (and also only available for DuckDB and Spark so far), but should mean you can do `clustering.to_csv('clusters.csv')`

**Reply by RossKen** [reply]
Good spot @ADBond, would you mind raising an issue for this to be added to the docs?

**Comment by jmacak-at-dl** [comment]
Well, shoot! That is exactly what I am looking for! Thank you so much!
